testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleExecuter.java

```diff
@@ -352,58 +352,32 @@ public void connect(PipedInputStream snk) throws IOException {
     TestDirectoryProvider getTestDirectoryProvider();
 
     /**
-     * Expects exactly one deprecation warning in the build output. If more than one warning is produced,
-     * or no warning is produced at all, the assertion fails.
+     * Expects the given deprecation warning.
      *
-     * @see #expectDeprecationWarnings(int)
-     * @deprecated Use {@link #expectDeprecationWarning(String)} instead.
+     * @implNote URLs to documentation should use /current/ as the version. This fixture will automatically replace it with the actual version tested.
      */
-    @Deprecated
-    GradleExecuter expectDeprecationWarning();
+    GradleExecuter expectDocumentedDeprecationWarning(String warning);
 
     /**
-     * Expects exactly the given deprecation warning.
+     * Do not call this method directly.
```

Comment:
🤔 

It seems this method is only needed in `AbstractGradleExecuter`. If so, we could have it there as a private/protected method.

Reply:
That was my thinking too, but the way we setup the executer uses the public interface.  I didn't want to do more surgery.